x-pack/plugins/apm/public/components/shared/redirect_with_offset/index.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocation, Redirect } from 'react-router-dom';
+import qs from 'query-string';
+import React from 'react';
+import moment from 'moment';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { getDateRange } from '../../../context/url_params_context/helpers';
+
+import {
+  TimeRangeComparisonEnum,
+  getSelectOptions,
+} from '../../../components/shared/time_comparison/get_comparison_options';
+
+export function RedirectWithOffset({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const location = useLocation();
+  const query = qs.parse(location.search);
+
+  const apmRouter = useApmRouter();
+  const matchingRoutes = apmRouter.getRoutesToMatch(location.pathname);
+  const matchesRoute = matchingRoutes.some((route) => {
+    return (
+      route.path === '/services' ||
+      route.path === '/traces' ||
+      route.path === '/service-map' ||
+      route.path === '/backends' ||
+      route.path === '/services/{serviceName}' ||
+      route.path === '/service-groups' ||
+      location.pathname === '/' ||
+      location.pathname === ''
+    );
```

Comment:
Maybe we should extract this into a function, something like isRouteWithTimeRange()? I think we're using it somewhere else too. I'm not a fan (I can say that because I wrote it) but I don't really have a better way of doing this right now.

Reply:
Good shout, indeed it's being used [here](https://github.com/elastic/kibana/blob/daea5a8519dafccdab851f7436941ef58a974ba7/x-pack/plugins/apm/public/components/shared/redirect_with_default_date_range/index.tsx#L34) as well. I can extract it to a function.

Reply:
Added in https://github.com/elastic/kibana/pull/127812/commits/9337e5d819fd2e594359bc93181b1459696d3e69